rerun_py/src/dataframe.rs

```diff
@@ -246,41 +249,74 @@ impl PyComponentColumnSelector {
 /// A type alias for any component-column-like object.
 #[derive(FromPyObject)]
 enum AnyColumn {
-    #[pyo3(transparent, annotation = "time_descriptor")]
-    TimeDescriptor(PyIndexColumnDescriptor),
-    #[pyo3(transparent, annotation = "time_selector")]
-    TimeSelector(PyIndexColumnSelector),
+    #[pyo3(transparent, annotation = "name")]
+    Name(String),
+    #[pyo3(transparent, annotation = "index_descriptor")]
+    IndexDescriptor(PyIndexColumnDescriptor),
+    #[pyo3(transparent, annotation = "index_selector")]
+    IndexSelector(PyIndexColumnSelector),
     #[pyo3(transparent, annotation = "component_descriptor")]
     ComponentDescriptor(PyComponentColumnDescriptor),
     #[pyo3(transparent, annotation = "component_selector")]
     ComponentSelector(PyComponentColumnSelector),
 }
 
 impl AnyColumn {
-    fn into_selector(self) -> ColumnSelector {
+    fn into_selector(self) -> PyResult<ColumnSelector> {
         match self {
-            Self::TimeDescriptor(desc) => ColumnDescriptor::Time(desc.0).into(),
-            Self::TimeSelector(selector) => selector.0.into(),
-            Self::ComponentDescriptor(desc) => ColumnDescriptor::Component(desc.0).into(),
-            Self::ComponentSelector(selector) => selector.0.into(),
+            Self::Name(name) => {
+                if !name.contains(':') && !name.contains('/') {
+                    Ok(ColumnSelector::Time(TimeColumnSelector {
+                        timeline: name.into(),
+                    }))
+                } else {
+                    let component_path =
+                        re_log_types::ComponentPath::from_str(&name).map_err(|err| {
+                            PyValueError::new_err(format!("Invalid component path '{name}': {err}"))
```

Comment:
```suggestion
                            PyValueError::new_err(format!("Invalid component path {name:?}: {err}"))
```

(so newlines etc get escaped)